### PR TITLE
Function

### PR DESCRIPTION
**Pull request by BaileyMatrascia**
# Merge function counter

### Type of Change

- [X] Feature
- [ ] Bug fix
- [X] Documentation

## Contributors

- @BaileyMatrascia 

## Reminder

All GitHub Actions should be in a passing state before any pull request is merged.

### CODE REVIEW

**Review by antlet — CHANGES_REQUESTED** [review]
The new function looks good, though since it is a smaller version of `match_funcdefs` I would suggest looking into/discussing with the group ideas for a better function name. This might mean renaming just the function itself or changing the naming conventions for the whole file.

The test case also looks good, but the name should also be changed when the function's name is. 

The big change that is lacking is the implementation of the feature into the CLI. A new command needs to be created that calls your function. 

One more potential change regards the user story for this task. The customer specifies wanting to "know how many functions are in a Python module or package.". This implies that either the module/package needs to be specified when finding the function count or the function needs to output the function counts for all modules/packages separately. You should ask the customer to specify about what they expect from this user story.

File: castanet/counter.py [hunk 1]
```diff
@@ -16,6 +16,21 @@ def match_imports(cast_dict):
     return imports_dictionary
 
 
+def match_function(cast_dict):
+    """ function to for counting the number of functions in a Python program"""
```
Comment (MaddyKapfhammer) [inline castanet/counter.py:20]: Docstring should not have a space between the quotations and the first letter. First letter should be capitalized and the doscstring should end with a period.

File: tests/test_counter.py [hunk 2]
```diff
@@ -103,3 +103,12 @@ def test_total_comment_returns_correct_number_comments():
     comment_dictionaries = {'say_hello.py': 1, '__init__.py': 3}
     number_comments = counter.sum_cast_dict(comment_dictionaries)
     assert number_comments == 4
+
+def test_match_function():
+    """Check that match_function identifies all of the comments in a directory."""
```
Comment (MaddyKapfhammer) [inline tests/test_counter.py:108]: I believe this docstring should be rewritten.

**Review by MaddyKapfhammer — APPROVED** [review]

**Review by antlet — DISMISSED** [review]
Looks great!

**Review by MaddyKapfhammer — APPROVED** [review]

**Review by antlet — APPROVED** [review]
Looks good

**Pull request merged**